test/integration/duplication/index.spec.ts

```diff
@@ -0,0 +1,58 @@
+import * as path from 'path';
+import { ToolpadEditor } from '../../models/ToolpadEditor';
+import { ToolpadHome } from '../../models/ToolpadHome';
+import { test, expect } from '../../playwright/test';
+import { readJsonFile } from '../../utils/fs';
+
+test('duplication', async ({ page, browserName }) => {
+  const dom = await readJsonFile(path.resolve(__dirname, './dom.json'));
+
+  const homeModel = new ToolpadHome(page);
+  await homeModel.goto();
+  const app = await homeModel.createApplication({ dom });
+
+  const editorModel = new ToolpadEditor(page, browserName);
+  await editorModel.goto(app.id);
+
+  {
+    await editorModel.openHierarchyMenu('connection');
+    const duplicateMenuItem = page.getByRole('menuitem', { name: 'Duplicate connection' });
+    await Promise.all([duplicateMenuItem.click(), page.waitForNavigation()]);
+
+    const input = page.getByLabel('base url');
+    await expect(input).toHaveValue('https://example.com/');
+
+    await editorModel.openHierarchyMenu('connection1');
```

Comment:
Would it be worth testing that the new menu item is added after duplicating, and removed after deleting?